todo_app/data/trello_items.py

```diff
@@ -6,35 +6,22 @@
 def process_full_trello_records():
     
     items_dict = api_request_get()
-    process_todo_items(items_dict)
-    process_completed_items(items_dict)    
+    process_tasks(items_dict)
+       
+      
+def process_tasks(items_dict):
+
+    for i in range(len(items_dict)):
+        list_name = items_dict[i]['name']
+        for card in items_dict[i]['cards']:
+            duplicate_task = False
+            for task in task_list:
+                if (card['id'] == task.id):
+                    duplicate_task = True
+                    task.status = list_name   
+            if (duplicate_task == False):
+                add_card_to_task_list(card, list_name)               
```

Comment:
You have introduced a lot of complexity and potential bugs by keeping a global `task_list` variable. If you wanted to make item names editable, you'd actually have to come and modify this block of code which is non-obvious and extra work.

Secondly, performance is not actually an issue with the number of tasks you have, but it would also be improved by simplifying your approach.

Much simpler approach would be something like this:
```suggestion
    task_list = []
    for trello_list in items_dict:
        list_name = trello_list['name']
        for card in trello_list['cards']:
            task_list.append(Item.from_trello_cards(card, list_name))
    return task_list            
```
And then use the return value of process_tasks rather than relying on a global variable.

(I also refactored the outer for-loop a bit to be simpler and nicer to read).

Another change you could make is `items_dict` is neither a dictionary nor just the items. It is a list of dictionaries, where each dictionary is a Trello "list" object.